### DISCUSSION TITLE
Feature request: render noexit nodes

### DISCUSSION THREAD

**Discussion by justwalked** [organicmaps/organicmaps#7890] [Ideas]
I would like to suggest rendering [noexit=yes](https://wiki.openstreetmap.org/wiki/Tag:noexit%3Dyes) nodes to let the user better understand which ways can and cannot be used to navigate.

Allow me to share my own story that led me writing this suggestion. I have been using OrganicMaps to navigate around Taiwan by walk. While not being perfect, OSM data is quite good in Taiwan and OrganicMaps does a good job at calculating itineraries, however I came across some unconnected ways or outdated information that made OrganicMaps route around entire areas I could have otherwise walked through. I soon learned that the data is not 100% perfect, especially in remote areas, so I have compensate for map limitations by guessing and asking locals.

So at some point I came across an area that can seemingly be crossed by two possible ways, either a small or a big road, but the small road is not connected so OrganicMaps always routes through the main road no matter what. As a pedestrian I wish I could take the small road to avoid dangerous traffic. So should I assume the way is incorrectly unconnected and go the small road? If it happens to be a dead end, I have to go back all the way, which means walking an extra 3.4 km for nothing, which is so frustrating for a pedestrian.

![map](https://github.com/organicmaps/organicmaps/assets/166682256/524ceea7-f6d6-4536-a2a1-d63b34ab7d1e)

[Link of the map above](https://www.openstreetmap.org/directions?engine=fossgis_osrm_car&route=22.72615%2C120.25867%3B22.72857%2C120.27195#map=17/22.72868/120.26586&layers=C)

On that day, I did not want to take risks so I walked the main road, and after crossing I saw that the small road seemed to be blocked by a big fence. I thought thank god I did not take the small road! Later, I wanted to edit the map to make it clearer that the small road is a dead end by adding a fence or something. To my surprise, I discovered there is a [noexit](https://www.openstreetmap.org/node/8512750885) attribute at the road end (also learned about noexit attribute), which is exactly the piece of information I *really* wished I had when I was on the road.

**Comment by patepelo** [comment]
Thank you very much, it's indeed a good idea. I'm glad you use OM frequently. I'll convert it right away into an issue since I see no reason not to implement this. Just be aware that it may take over a year to be implemented since there is a pile of features of over 2k issues and little free time from contributors. 

Do you have some proposals on how it could be represented? Maybe a black dot at the ending node? Or a no-exit sign like in the Id editor?

**Reply by biodranik** [reply]
Maybe like this?
1. ![выява](https://github.com/organicmaps/organicmaps/assets/170263/eb18e6ec-c586-48ff-8121-fb0412e64c9b)
2. ![выява](https://github.com/organicmaps/organicmaps/assets/170263/f2f76e81-57d4-43f8-9e75-e73c7eadc41a)
